tfjs-backend-webgpu/src/backend_webgpu.ts

```diff
@@ -453,11 +463,26 @@ export class WebGPUBackend extends KernelBackend {
 
     const encoder = this.device.createCommandEncoder();
     const pass = encoder.beginComputePass();
+    if (shouldTimeProgram) {
+      if (this.supportTimeQuery) {
+        pass.writeTimestamp(this.querySet, 0);
+      } else {
+        query = this.startTimer();
+      }
+    }
     pass.setPipeline(pipeline);
     pass.setBindGroup(0, bg);
     pass.dispatch(
         program.dispatch[0], program.dispatch[1], program.dispatch[2]);
+    if (shouldTimeProgram) {
+      if (this.supportTimeQuery) {
+        pass.writeTimestamp(this.querySet, 1);
+      } else {
+        query = this.endTimer(query);
```

Comment:
If I'm reading this right, it doesn't make sense to endTimer here; nothing has actually happened between startTimer and endTimer except a few encoding commands.

Even the old code doesn't make much sense as it contains the submit, but it doesn't wait for anything to happen.
I think you should consider removing the CPU timing entirely. Or, keeping it as a separate timing stat that's always enabled - not an alternative to GPU timing.